crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -318,7 +318,19 @@ impl ShaderCache {
                 render_device
                     .wgpu_device()
                     .push_error_scope(wgpu::ErrorFilter::Validation);
-                let shader_module = render_device.create_shader_module(module_descriptor);
+
+                // `validate_shader` is evaluated to determine whether or not to perform
+                //  validation checks
+                let validate_shader = shader.validate_shader.clone();
```

Comment:
This shouldn't be needed: you can just make your little config enums `Copy`. Or in cases where that's more expensive, match on `&validate_shader` or `shader.validate_shader`.